### ISSUE TITLE
Info on how to integrate DataLad in containerized workflows/pipelines: maybe or madness?

### ISSUE THREAD

**Issue by PeerHerholz** [datalad-handbook/book#652]
**Is your feature request related to a problem? Please describe.**
Nope, not a problem, but rather an idea/discussion. While we have `datalad containers` to track what's happening when utilizing `containers` to analyze data, I was wondering re use cases where `DataLad` "automatically" runs within the `container` as part of the pipeline/workflow, placing outputs under version control. For example, `heudiconv` has an option to create a `DataLad` dataset from the converted files or add them to an existing one.  

**Describe the solution you'd like**
At this point maybe a conversation regarding the applicability of automated `DataLad` fun within containerized pipelines/workflows besides `sourcedata` to `rawdata` conversion.  

**Describe alternatives you've considered**
As far as I can tell, `datalad containers` is preferrable over automated `DataLad` functionality. Thus, I wanted to start the conversation on potential exceptions (given that they exist at all).

**Additional context**
`DataLad` = love.

**Comment by adswa** [comment]
Let me see if I get this correct: You envision a container with any workflow to be used not necessarily inside a dataset, and the workflow contains DataLad to either
- take whatever data it is ran on, transform it into a dataset, perform the workflow, and save the results in the same or a new dataset, or
- perform the workflow, and save the results in a new dataset

Basically: Input = arbitrary, analysis = arbitrary, output = dataset?

**Comment by PeerHerholz** [comment]
Yeah, I think so. The problems/issues I had so far were related to "feature fallacy" and thus I wanted to start a conversation on this. One thing is that we simply can't and shouldn't provide support for every possible use case, the other thing is that certain `provenance` and `version control` principles should be encouraged/(slightly) enforced.  I think the options you mentioned sound reasonable and would help a lot of folks. However, I'm not sure how these things should be positioned wrt `datalad containers`.

**Comment by mih** [comment]
Recent work made me experience this first hand. I think the encapsulation approach is a key issue that is also not that obvious.

Right now I think that using datalad inside a container that aims to improve reproducibility. This is not because of an immaturity issue with datalad itself, but is a consequence of the fact that datalad is the interface to the evolving world around a past snapshot of some what. That interface must remain flexible (e.g. learn new features to access new data sources), because they may become a requirement over time (e.g., because the needed data is hosted with technology that did not exist when the container snapshot was made).
